code/src/Components/Details.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable react/jsx-indent-props */
+/* eslint-disable indent */
+import React, { useState, useEffect } from 'react';
+import { useParams, Link } from 'react-router-dom';
+import './Details.css';
+import './List.css';
+
+export const Details = () => {
+    const { id } = useParams();
+    const [movie, setMovie] = useState();
+    // The `useEffect` hook is being passed two arguments: a function and an array of
+    // dependencies. The function in this case is responsible for the fetch request and
+    // updating the movie state. The array of dependencies contains a single element,
+    // `id`, meaning that the function will only be triggered when the `id` value
+    // changes. //
+    useEffect(() => {
+        fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=986aaac98bfe3b83f202c023b975310e&language=en-US`)
```

Comment:
The` Details` component is using an API key directly in the fetch URL. This is generally not a good practice as it exposes the API key publicly. It's better to keep the key in a `.env` file and access it through an environment variable.